### PR TITLE
[838] Add the ability to edit other qualifications to support 

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

We are adding functionality to the support console to update A levels and other qualifications. 

## Changes proposed in this pull request

- Change the location of the change link to reflect the fact that it can change more than just the year and grade. 
- Add a single page which can update all the values 
- This largely follows very similar to logic to the page on the candidate interface with a few minor differences.

## Screenshots

image

image

image

image

## Guidance to review

- Do we want the `continuous applications` label on this PR?

- For another potential PR but there is another section which can be chosen by the user but can't be edited: `Country or territory where you studied`

- Are we happy for the `Other UK qualification` not to autocomplete? Potenitally a further iteration can be to add that.

## Link to Trello card

https://trello.com/c/4utTkWmJ/838-support-edit-other-qualifications-functionality

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by avinhurry** [comment]
[Support: Edit "other qualifications" functionality](https://trello.com/c/4utTkWmJ/838-support-edit-other-qualifications-functionality)

File: app/components/shared/qualification_row_component.html.erb [hunk 1]
```diff
@@ -10,18 +10,13 @@
...
-    <% if editable %>
-      <%= govuk_link_to support_interface_application_form_edit_other_qualification_grade_path(qualification_id: qualification.id) do %>
-        Change <span class="govuk-visually-hidden">grade</span>
-      <% end %>
-    <% end %>
+  </td>
+  <td class="govuk-table__cell">
+  <% if editable %>
+    <%= govuk_link_to('Change', support_interface_application_form_edit_other_qualification_path(id: qualification.id)) %>
+  <% end %>
```
Comment (tomas-stefano) [inline app/components/shared/qualification_row_component.html.erb:20]: Is this shared component used in what interfaces? Provider and support?

**Review by tomas-stefano — APPROVED** [review]
Minor question. Code looks good to me. Will somebody test this on a review app?

File: app/components/shared/qualification_row_component.html.erb (hunk 1 above)
Comment (avinhurry) [reply app/components/shared/qualification_row_component.html.erb:20]: Yes, provider and support. There's a comment at the top of the file: https://github.com/DFE-Digital/apply-for-teacher-training/blob/3a54475e19196fce8e9d89d8327699b6314a472f/app/components/shared/qualifications_table_component.rb#L1

**Comment by avinhurry** [comment]
> Minor question. Code looks good to me. Will somebody test this on a review app?

Thanks @tomas-stefano. Yes, will tag now.

**Review by lsalivonaite-transform — APPROVED** [review]
All good, just noticed 2 things, that there are not validations on Qualification name and Grade on 'Qualifications Outside of UK' and 'Other UK Qualifications'.
Screenshot 2023-10-30 at 17 03 05
Screenshot 2023-10-30 at 17 03 13

**Comment by avinhurry** [comment]
> All good, just noticed 2 things, that there are not validations on Qualification name and Grade on 'Qualifications Outside of UK' and 'Other UK Qualifications'. Screenshot 2023-10-30 at 17 03 05 Screenshot 2023-10-30 at 17 03 13

Thanks @lsalivonaite-transform! I've added the presence validation to other uk and non uk when those radios are selected.

For grade, we don't validate on the candidate interface when it's other uk and non uk.

**Comment by MylesJarvis** [comment]
I was writing this comment but we merged the PR before I got it out:

* This may be entirely unrelated (given what I'm about to say is another section that isn't other qualifications), but when testing on here vs prod, if I try and click a change link for work history, I get a page reload and jump back to the top of the page.

I'll test on QA/Prod once it's gone through the pipeline, as I can no longer test on review app.

**Pull request merged**